src/org/thoughtcrime/securesms/ConversationItem.java

```diff
@@ -500,15 +503,44 @@ private void setContactPhoto() {
   }
 
   private SpannableString linkifyMessageBody(SpannableString messageBody, boolean shouldLinkifyAllLinks) {
-    boolean hasLinks = Linkify.addLinks(messageBody,
-        shouldLinkifyAllLinks ? Linkify.EMAIL_ADDRESSES|Linkify.WEB_URLS|Linkify.PHONE_NUMBERS : 0);
+    if (!shouldLinkifyAllLinks) {
+      return messageBody;
+    }
+    boolean hasLinks = false;
+
+    Pattern cmdPattern = Pattern.compile("(?<=^|\\s)/[a-zA-Z][a-zA-Z@\\d_/.-]{0,254}");
+    hasLinks = Linkify.addLinks(messageBody, cmdPattern, "cmd:", null, null) || hasLinks;
+
+    Linkify.TransformFilter urlFilter = (match, url) -> {
+      if (url.startsWith("http://") || url.startsWith("https://")) {
+        return url;
+      } else {
+        return "http://" + url;
```

Comment:
shouldn't we force https here?

Reply:
http is the standard and if you type `github.com` then your browser will use http. (you will then be redirected to https://github.com)

So, technically http is correct but as very few sites still use http, I agree with you, we should actually use https, at the cost of breaking a very little percentage of links.

@link2xt @adbenitez what do you think?